kernel/src/schema.rs

```diff
@@ -167,6 +178,16 @@ impl StructField {
         &self.metadata
     }
 
+    /// Convert our metadata into a HashMap<String, String>. Note this copies all the data so can be
+    /// expensive for large metadata
+    pub fn metadata_as_string(&self) -> HashMap<String, String> {
```

Comment:
super mega nit but maybe rename to `metadata_to_strings` (or perhaps `metadata_to_map`) to (1) communicate similar to `.to_string()` that it is expensive, and (2) clarify this is `to_strings` and not `to_string` which would imply `String` type?